leader/leader_test.go

```diff
@@ -0,0 +1,213 @@
+						Name:      "leader-test",
+						Namespace: "testns",
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "leader-test",
+							},
+						},
+					},
+				},
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "leader-test",
+						Namespace: "testns",
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "leader-test",
+							},
+						},
+					},
+				},
+			)
+		})
+		It("should return an error when POD_NAME is not set", func() {
+			os.Unsetenv("POD_NAME")
+			err := Become(context.TODO(), "leader-test")
+			Expect(err).ShouldNot(BeNil())
+		})
+		It("should not return an error", func() {
+			os.Setenv("POD_NAME", "leader-test")
+			nsFile, err := setupNamespace("testns")
+			if err != nil {
+				Fail(err.Error())
+			}
+			defer os.Remove(nsFile.Name())
+			readNamespace = func() ([]byte, error) {
+				return ioutil.ReadFile(nsFile.Name())
+			}
+
+			err = Become(context.TODO(), "leader-test", WithClient(client))
+			Expect(err).Should(BeNil())
+		})
+	})
+	Describe("isPodEvicted", func() {
+		var (
+			leaderPod *corev1.Pod
+		)
+		BeforeEach(func() {
+			leaderPod = &corev1.Pod{}
+		})
+		It("should return false with an empty status", func() {
+			Expect(isPodEvicted(*leaderPod)).To(Equal(false))
+		})
+		It("should return false if reason is incorrect", func() {
+			leaderPod.Status.Phase = corev1.PodFailed
+			leaderPod.Status.Reason = "invalid"
+			Expect(isPodEvicted(*leaderPod)).To(Equal(false))
+		})
+		It("should return false if pod is in the wrong phase", func() {
+			leaderPod.Status.Phase = corev1.PodRunning
+			Expect(isPodEvicted(*leaderPod)).To(Equal(false))
+		})
+		It("should return true when Phase and Reason are set", func() {
+			leaderPod.Status.Phase = corev1.PodFailed
+			leaderPod.Status.Reason = "Evicted"
+			Expect(isPodEvicted(*leaderPod)).To(Equal(true))
+		})
+	})
+	Describe("getOperatorNamespace", func() {
+		It("should return error when namespace not found", func() {
+			namespace, err := getOperatorNamespace()
+			Expect(err).To(Equal(ErrNoNamespace))
+			Expect(namespace).To(Equal(""))
+		})
+		It("should return namespace", func() {
+
+			nsFile, err := setupNamespace("testnamespace")
+			if err != nil {
+				Fail(err.Error())
+			}
+			defer os.Remove(nsFile.Name())
+			readNamespace = func() ([]byte, error) {
+				return ioutil.ReadFile(nsFile.Name())
+			}
+
+			// test
+			namespace, err := getOperatorNamespace()
+			Expect(err).Should(BeNil())
+			Expect(namespace).To(Equal("testnamespace"))
+		})
+		It("should trim whitespace from namespace", func() {
+
+			nsFile, err := setupNamespace("   testnamespace	   ")
+			if err != nil {
+				Fail(err.Error())
+			}
+			defer os.Remove(nsFile.Name())
+			readNamespace = func() ([]byte, error) {
+				return ioutil.ReadFile(nsFile.Name())
+			}
```

Comment:
This can be simplified to:

```go
readNamespace = func() ([]byte, err) {
    return []byte("   testnamespace    "), nil
}
```

Reply:
Oh man not sure how I missed that. Thanks